src/Microsoft.AspNetCore.Server.IISIntegration/Server/IISServerSetupFilter.cs

```diff
@@ -20,6 +22,12 @@ public Action<IApplicationBuilder> Configure(Action<IApplicationBuilder> next)
         {
             return app =>
             {
+                var server = app.ApplicationServices.GetService<IServer>();
+                if (server?.GetType() != typeof(IISHttpServer))
+                {
+                    throw new InvalidOperationException("Application is running inside IIS process but is not configured to use IIS server. Call UseIISIntegration after adding all other servers.");
```

Comment:
Hold off on merging this until we decide what the correct guidance is.

Reply:
We can merge this PR, but don't mention the second sentence for now, correct?